From 289685a9b369bda990c4f22bbc7659ad492e6dbb Mon Sep 17 00:00:00 2001
From: Stefan Wahren <stefan.wahren@i2se.com>
Date: Sun, 3 Feb 2019 14:02:34 +0100
Subject: [PATCH] pinctrl: bcm2835: Direct GPIO config changes to
 generic pinctrl

commit b6e5531c0f80de2779c87d0235b4fde5310a83b5 upstream.

In order to support GPIO config changes direct these to the generic pinctrl.
This also requires an adjust of the return code for unsupported parameter
otherwise gpiod_configure_flags wont work as expected.

Signed-off-by: Stefan Wahren <stefan.wahren@i2se.com>
Signed-off-by: Linus Walleij <linus.walleij@linaro.org>
---
 drivers/pinctrl/bcm/pinctrl-bcm2835.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/drivers/pinctrl/bcm/pinctrl-bcm2835.c
+++ b/drivers/pinctrl/bcm/pinctrl-bcm2835.c
@@ -342,6 +342,7 @@ static const struct gpio_chip bcm2835_gp
 	.get = bcm2835_gpio_get,
 	.set = bcm2835_gpio_set,
 	.base = 0,
+	.set_config = gpiochip_generic_config,
 	.ngpio = BCM2835_NUM_GPIOS,
 	.can_sleep = false,
 };
@@ -960,7 +961,7 @@ static int bcm2835_pinconf_set(struct pi
 			break;
 
 		default:
-			return -EINVAL;
+			return -ENOTSUPP;
 
 		} /* switch param type */
 	} /* for each config */
